sda/cmd/s3inbox/proxy.go

```diff
@@ -511,6 +550,31 @@ func (p *Proxy) requestInfo(fullPath string) (string, int64, error) {
 
 }
 
+// checkFileExists makes a request to the S3 to check whether the file already
+// is uploaded. Returns a bool indicating whether the file was found.
+func (p *Proxy) checkFileExists(fullPath string) (bool, error) {
+	filePath := strings.Replace(fullPath, "/"+p.s3.Bucket+"/", "", 1)
+	client, err := storage.NewS3Client(p.s3)
+	if err != nil {
+		log.Warnf("could not connect to s3: %v", err.Error())
+
+		return false, err
+	}
+
+	result, err := client.HeadObject(context.TODO(), &s3.HeadObjectInput{
+		Bucket: &p.s3.Bucket,
+		Key:    &filePath,
+	})
+	if err != nil {
+		// typically 404, if file is not found
+		log.Debugf("could not find file: %v", err)
+
+		return false, nil
+	}
+
+	return result != nil, nil
```

Comment:
```suggestion

	return result != nil, err
```

This can be shortened to this since the check for error comes first.

Reply:
But that way we would forward error messages like "file not found". I would expect the function to return `false, nil` in case the file is not found, not `false, <an error including 404>`. Would you agree?
(It is also worth mentioning that those errors from s3 are hard to parse, as they don't have `err.Error()`.)

Reply:
This is a bit of semantics, where do you parse the error, in the function or at the receiving end?

IF this function returns the raw error we can easily see them in the static code tests and implement the handling where the function is being called from.